### ISSUE TITLE
Search with unicode characters in menubar

### ISSUE THREAD

**Issue by p-himik** [awesomeWM/awesome#1420]
Output of `awesome --version`:

```
awesome v4.0 (Harder, Better, Faster, Stronger)
 • Compiled against Lua 5.2.4 (running with Lua 5.2)
 • D-Bus support: ✔
 • execinfo support: ✔
 • RandR 1.5 support: ✘
 • LGI version: 0.9.0
```

**How to reproduce the issue:**
1. open menubar
2. type in some non-ascii characters
3. start pressing backspace

**Actual result:**
Some backspace presses don't remove anything.
I guess there's a single byte removed from a multi-byte character, but the visual representation doesn't change.

**Expected result:**
One backspace press should remove exactly one character.

**Comment by psychon** [comment]
I think we have some hack in awful.prompt that assumes that 'certain kinds of errors' are due to multi-byte codepoints and then just deletes more bytes to handle this. That's of course an ugly hack.

The "proper" thing to do would be to reinvent the [GtkEntry](https://developer.gnome.org/gtk3/stable/GtkEntry.html)-wheel and do all of this properly. For example, [this](https://git.gnome.org/browse/gtk+/tree/gtk/gtkentry.c?id=f4afa000a14d7b53a8efdc0782c37576cbf6c3b0#n5385) should be the code that handles backspace.

Sigh. :-(

**Comment by actionless** [comment]
mb to use gtk entry via lgi? :)

**Comment by Elv13** [comment]
> mb to use gtk entry via lgi? :)

It isn't possible. Using GTK from the same thread as Awesome creates potential deadlocks.

**Comment by p-himik** [comment]
Maybe there's some way to make Awesome always use some predefined language for the menu?
If not, is it possible to make Awesome have its own keyboard layout that does not depend on the current client? I use kbdd to make sure all X clients can have their own keyboard layout, so maybe there's some way to make the menu behave like a client in this regard.

**Comment by psychon** [comment]
...how is this issue related to keyboard layout? What do you mean with "language for the menu"?

**Comment by p-himik** [comment]
Suppose I have two clients and kbdd running. In this setup, I can make client 1 have ru_RU layout and client 2 have en_US layout.
Now, when client 2 is in focus, typing in Awesome menu uses en_US layout. And when client 1 is in focus, typing in Awesome menu uses ru_RU layout. Since I always have some clients with ru_RU layout active, it becomes really annoying to accidentally type in Russian in the Awesome menu just to have hold backspace for a few seconds.
I was wondering if there's a way to make Awesome always accept only single layout (en_US would be my personal choice) or make it not use the layout of the currently focused client but rather its own.

**Comment by p-himik** [comment]
Seems like it got fixed along with #3308.
